src/observability/consumer.go

```diff
@@ -83,10 +83,6 @@ func validateSummaryRequest(consumer *SummaryConsumer, summary types.SystemSumma
 		return false
 	}
 
-	if consumer.Deployment != "" && consumer.Deployment != summary.Deployment {
```

Comment:
If we're removing the `Deployment` field, shouldn't we use the data in `Workload.Name` to validate this? Check if removing the `Deployment` field would affect `karmor summary` in anyway.

Reply:
@seswarrajan WDYT? Are we depend on `Deployment` field for any case?